Civi.php

```diff
@@ -95,6 +95,7 @@ public static function service($id) {
   public static function reset() {
     self::$statics = array();
     Civi\Core\Container::singleton();
+    \Civi::service('settings_manager')->useDefaults();
```

Comment:
Agree that putting `useDefaults()` in `boot()` would have unintended consequences. IIRC, during dev, I'd originally applied defaults during `boot()` but had to move it. (Maybe to support edge cases around using `hook_civicrm_config` to manipulate `$civicrm_setting` data?)

FWIW, in my mind, there are two different notions of "reset":
 * _Thin reset_: Tear down running service objects... basically just flush statics. This was a new function, `Civi::reset()`. It leaves the system in an unbootstrapped state. (FWIW, in a design like "Symfony Standard Edition", that's a pretty much all you need to do. Much of the initialization is sequenced automatically/implicitly/on-demand when you request things from the container.)
 * _Full reset_: Tear down _and restart_ the services. This was `CRM_Core_Config::singleton(1,1)`.  It leaves the system in a bootstrapped state. This was the preexisting functionality used by `CiviUnitTestCase` (and takes a lot of care to preserve sequence of steps).

FWIW, the current `CiviUnitTestCase` says this:

```php
    \Civi::reset();
    $config = CRM_Core_Config::singleton(TRUE, TRUE); // ugh, performance
```

My gut says this patch is more intellectually consistent:

```diff
diff --git a/Civi/Test/CiviTestListener.php b/Civi/Test/CiviTestListener.php
index 3bd6f0d..c87f60f 100644
--- a/Civi/Test/CiviTestListener.php
+++ b/Civi/Test/CiviTestListener.php
@@ -98,6 +98,7 @@ class CiviTestListener extends \PHPUnit_Framework_BaseTestListener {
     $config = \CRM_Core_Config::singleton(TRUE, TRUE); // ugh, performance
     \CRM_Utils_System::flushCache();
     \Civi::reset();
+    \CRM_Core_Config::singleton(TRUE, TRUE);
     \CRM_Core_Session::singleton()->set('userID', NULL);

     if (property_exists($config->userPermissionClass, 'permissions')) {
```

although (either way) we probably want to spot-check a couple other extensions to see that they still run tests correctly.